x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/47_semantic_text_knn.yml

```diff
@@ -0,0 +1,379 @@
+
+  - do:
+      inference.put:
+        task_type: text_embedding
+        inference_id: dense-inference-id
+        body: >
+          {
+            "service": "text_embedding_test_service",
+            "service_settings": {
+              "model": "my_model",
+              "dimensions": 10,
+              "similarity": "cosine",
+              "api_key": "abc64"
+            },
+            "task_settings": {
+            }
+          }
+
+  - do:
+      inference.put:
+        task_type: text_embedding
+        inference_id: dense-inference-id-2
+        body: >
+          {
+            "service": "text_embedding_test_service",
+            "service_settings": {
+              "model": "my_model",
+              "dimensions": 10,
+              "similarity": "cosine",
+              "api_key": "abc64"
+            },
+            "task_settings": {
+            }
+          }
+
+  - do:
+      indices.create:
+        index: test-semantic-text-index
+        body:
+          mappings:
+            properties:
+              inference_field:
+                type: semantic_text
+                inference_id: dense-inference-id
+
+  - do:
+      indices.create:
+        index: test-semantic-text-index-2
+        body:
+          mappings:
+            properties:
+              inference_field:
+                type: semantic_text
+                inference_id: dense-inference-id-2
+
+  - do:
+      indices.create:
+        index: test-dense-vector-index
+        body:
+          mappings:
+            properties:
+              inference_field:
+                type: dense_vector
+                dims: 10
+                similarity: cosine
+
+  - do:
+      indices.create:
+        index: test-incompatible-dense-vector-index
+        body:
+          mappings:
+            properties:
+              inference_field:
+                type: dense_vector
+                dims: 3
+                similarity: cosine
+
+  - do:
+      index:
+        index: test-semantic-text-index
+        id: doc_1
+        body:
+          inference_field: [ "inference test", "another inference test" ]
+        refresh: true
+
+  - do:
+      index:
+        index: test-semantic-text-index-2
+        id: doc_2
+        body:
+          inference_field: [ "inference test", "another inference test" ]
+        refresh: true
+
+  - do:
+      index:
+        index: test-dense-vector-index
+        id: doc_3
+        body:
+          inference_field: [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 ]
+        refresh: true
+
+  - do:
+      index:
+        index: test-incompatible-dense-vector-index
+        id: doc_4
+        body:
+          inference_field: [ 1, 2, 3 ]
+        refresh: true
+
+---
+"Nested knn queries using the old format on semantic_text embeddings and inference still work":
+  - skip:
+      features: [ "headers" ]
+
+  - do:
+      headers:
+        # Force JSON content type so that we use a parser that interprets the floating-point score as a double
```

Comment:
Nit - I don't think we need to force JSON in these tests, as we're not looking into scores...

Reply:
You caught me copy-pasta-ing. 😂 Good catch, tested and can be removed.